src/com/palmergames/bukkit/towny/huds/MapHUD.java

```diff
@@ -0,0 +1,161 @@
+import com.palmergames.bukkit.towny.object.TownyWorld;
+import com.palmergames.bukkit.towny.object.Translation;
+import com.palmergames.bukkit.towny.object.WorldCoord;
+import com.palmergames.bukkit.util.Colors;
+import org.bukkit.Bukkit;
+import org.bukkit.ChatColor;
+import org.bukkit.entity.Player;
+import org.bukkit.scoreboard.DisplaySlot;
+import org.bukkit.scoreboard.Objective;
+import org.bukkit.scoreboard.Scoreboard;
+
+public class MapHUD {
+	private static int lineWidth = 20, lineHeight = 10;
+	
+	public static void toggleOn(Player player) {
+		Scoreboard board = Bukkit.getScoreboardManager().getNewScoreboard();
+		Objective objective = board.registerNewObjective("MAP_HUD_OBJ", "", "");
+		objective.setDisplaySlot(DisplaySlot.SIDEBAR);
+		
+		int score = lineHeight + 2;
+		for (int i = 0; i < lineHeight; i++) {
+			board.registerNewTeam("mapTeam" + i).addEntry(ChatColor.values()[i].toString());
+			objective.getScore(ChatColor.values()[i].toString()).setScore(score);
+			score--;
+		}
+		
+		String townEntry = ChatColor.DARK_GREEN + Translation.of("town_sing") + ": ";
+		String ownerEntry = ChatColor.DARK_GREEN + Translation.of("owner_status") + ": ";
+		
+		board.registerNewTeam("townTeam").addEntry(townEntry);
+		objective.getScore(townEntry).setScore(2);
+		
+		board.registerNewTeam("ownerTeam").addEntry(ownerEntry);
+		objective.getScore(ownerEntry).setScore(1);
+		
+		player.setScoreboard(board);
+		updateMap(player);
+	}
+	
+	public static void updateMap(Player player) {
+		updateMap(player, WorldCoord.parseWorldCoord(player));
+	}
+	
+	public static void updateMap(Player player, WorldCoord wc) {
+		Scoreboard board = player.getScoreboard();
+		if (board == null) {
+			toggleOn(player);
+			return;
+		} else if (board.getObjective("MAP_HUD_OBJ") == null) {
+			HUDManager.toggleOff(player);
+			return;
+		}
+
+		Objective objective = board.getObjective("MAP_HUD_OBJ");
+		objective.setDisplayName(ChatColor.GOLD + "Towny Map " + ChatColor.WHITE + "(" + wc.getX() + ", " + wc.getZ() + ")");
+
+		TownyWorld world;
+		try {
+			world = wc.getTownyWorld();
+			if (!world.isUsingTowny())
+				throw new NotRegisteredException();
+		} catch (NotRegisteredException e) {
+			//Toggle off if towny world is not registered or not using towny.
+			HUDManager.toggleOff(player);
+			return;
+		}
+		
+		Resident resident = TownyAPI.getInstance().getResident(player.getName());
+		boolean hasTown = resident.hasTown();
+
+		int halfLineWidth = lineWidth/2;
+		int halfLineHeight = lineHeight/2;
+		
+		String[][] map = new String[lineWidth][lineHeight];
+		int x, y = 0;
+		for (int tby = wc.getX() + (lineWidth - halfLineWidth - 1); tby >= wc.getX() - halfLineWidth; tby--) {
+			x = 0;
+			for (int tbx = wc.getZ() - halfLineHeight; tbx <= wc.getZ() + (lineHeight - halfLineHeight - 1); tbx++) {
+				try {
+					TownBlock townblock = world.getTownBlock(tby+1, tbx);
+					if (!townblock.hasTown())
+						throw new TownyException();
```

Comment:
I feel like throwing an exception is a redundant overhead when you can just perform the logic inside the if-statement.